### PR TITLE
Memory Aware Synapses on TinyImageNet

### PR DESCRIPTION
**Pull request by rmassidda**
This pull request introduces a script to replicate the analysis of the Memory Aware Synapses strategy (ContinualAI/avalanche#945) on the TinyImageNet dataset. The experimental setting is described within "A continual learning survey: Defying forgetting in classification tasks" (De Lange et al., [link](https://doi.org/10.1109/TPAMI.2021.3057446)).

The implementation is modeled after the `test_stinyimagenet` method for the LwF strategy, which replicates the same setting from the same paper. Thus, it adopts the `MultiHeadVGGSmall` model over ten experiences of the `SplitTinyImageNet` benchmark.

### CODE REVIEW

File: strategies/lwf/experiment.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def test_pmnist(self, override_args=None):
         if args.check and target_acc > avg_stream_acc:
             self.assertAlmostEqual(target_acc, avg_stream_acc, delta=0.02)
 
-    def test_stinyimagenet(self, override_args=None):
+    def test_stinyimagenet(self, override_args=None, dataset_root=None):
```
Comment (AndreaCossu) [inline strategies/lwf/experiment.py:103]: can you move the `dataset_root` argument inside the dictionary of arguments inside the function?

File: strategies/mas/experiment.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+class MAS(unittest.TestCase):
+    """
+    Reproducing Memory Aware Synapses experiments from paper
+    "A continual learning survey: Defying forgetting in classification tasks"
+    by De Lange et al.
+    https://doi.org/10.1109/TPAMI.2021.3057446
+    """
+
+    def test_stinyimagenet(self, override_args=None, dataset_root=None):
```
Comment (AndreaCossu) [inline strategies/mas/experiment.py:27]: same here for `dataset_root` argument. This is just to enforce the same interface for all tests

File: strategies/mas/experiment.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            target_acc = target[0]
+        else:
+            target_acc = target
+        target_acc = float(target_acc)
+
+        print(f"The target value was {target_acc:.2f}")
+
+        # Check if the result is close to the target
+        if args.check and target_acc > avg_stream_acc:
+            self.assertAlmostEqual(target_acc, avg_stream_acc, delta=0.02)
```
Comment (AndreaCossu) [inline strategies/mas/experiment.py:97]: can you change this to the new default 0.03?

**Review by AndreaCossu — COMMENTED** [review]
Thanks for the PR, looks great! I added few minor comments. Once you adjusted those I'll merge the PR :smile:

**Comment by rmassidda** [comment]
Ok, the last commits addressed them!

**Comment by AndreaCossu** [comment]
Thanks!

**Review by AndreaCossu — APPROVED** [review]

**Pull request merged**